src/fs/service.go

```diff
@@ -650,3 +677,60 @@ func (service *MercuryFsService) uploadFile(writer http.ResponseWriter, request
 
 	return
 }
+
+// rename the given file path
+func renameFile(p string) string {
+	//get the file suffix
+	ext := path.Ext(p)
+	//get the filename without suffix
+	baseName := strings.Replace(p, ext, "", 1)
+
+	timeStamp := time.Now().Format("20060102-1504")
+
+	return baseName + "-" + timeStamp + ext
+}
+
+/*
+status:
+0 => file not exists,
+1 => file exists,
+2 => file exists and have the same md5
```

Comment:
You can make an enum for these. Make a custom int type and use iota for assigning them values automatically.
Ref: https://blog.learngoprogramming.com/golang-const-type-enums-iota-bc4befd096d3